connector/connect/client/jvm/src/main/scala/org/apache/spark/sql/SparkSession.scala

```diff
@@ -667,6 +667,7 @@ class SparkSession private[sql] (
    * @since 3.4.0
    */
   override def close(): Unit = {
+    client.releaseSession()
```

Comment:
Should releaseSession be called inside `client.shutdown()`?

Reply:
Since client is an internal API that shouldn't be used to end user, I deliberately left it to the session layer to call release session. Here this allows me to test some internal behaviours with the client in the tests. In the future, I imagine this could be useful in some situations where we may need to exchange a client / get a new GRPC channel without closing the session.